src/data_model/eng2math.jl

```diff
@@ -424,12 +421,15 @@ function _map_eng2math_switch!(data_math::Dict{String,<:Any}, data_eng::Dict{<:A
         # if !all(isapprox.(get(eng_obj, "rs", zeros(1, 1)), 0)) && !all(isapprox.(get(eng_obj, "xs", zeros(1, 1)), 0)) # TODO enable real switches
             # build virtual bus
 
-            f_bus = data_math["bus"]["$(math_obj["f_bus"])"]
+            f_bus = deepcopy(data_math["bus"]["$(math_obj["f_bus"])"])
+            t_bus = deepcopy(data_math["bus"]["$(math_obj["t_bus"])"])
 
             bus_obj = Dict{String,Any}(
                 "name" => "_virtual_bus.switch.$name",
                 "bus_i" => length(data_math["bus"])+1,
                 "bus_type" => get(eng_obj, "state", CLOSED) == OPEN ? 4 : 1,
+                "terminals" => t_bus["terminals"],  # connected to the switch on the to-side
+                "grounded" => t_bus["grounded"],  # connected to the switch on the to-side
```

Comment:
Is it correct that you just replace the switch with a branch and do not actually use the virtual bus for now? If so, I will ignore this for now, and come back to it when you actually add in the virtual bus and switch.

Reply:
yes, the virtual bus is not currently in use, its definition is in place for when we formally introduce switches to the constraints